src/zenml/mlstacks/utils.py

```diff
@@ -0,0 +1,525 @@
+
+    Returns:
+        A dictionary of tags.
+    """
+    return dict(tag.split("=") for tag in tags) if tags else {}
+
+
+@verify_installation
+def _construct_stack(params: Dict[str, Any]) -> "Stack":
+    """Constructs mlstacks `Stack` object from raw Click CLI params.
+
+    Args:
+        params: Raw Click CLI params.
+
+    Returns:
+        A mlstacks `Stack` object.
+    """
+    from mlstacks.models import Stack
+
+    tags = _get_stack_tags(params["tags"])
+
+    return Stack(
+        spec_version=1,
+        spec_type="stack",
+        name=params["stack_name"],
+        provider=params["provider"],
+        default_region=params["region"],
+        default_tags=tags,
+        components=[],
+    )
+
+
+@verify_installation
+def convert_click_params_to_mlstacks_primitives(
+    params: Dict[str, Any],
+    zenml_component_deploy: bool = False,
+) -> Tuple["Stack", List["Component"]]:
+    """Converts raw Click CLI params to mlstacks primitives.
+
+    Args:
+        params: Raw Click CLI params.
+        zenml_component_deploy: A boolean indicating whether the stack
+            contains a single component and is being deployed through `zenml
+            component deploy`
+
+    Returns:
+        A tuple of Stack and List[Component] objects.
+    """
+    from mlstacks.constants import MLSTACKS_PACKAGE_NAME
+    from mlstacks.models import Component, Stack
+
+    stack: Stack = _construct_stack(params)
+    components: List[Component] = _construct_components(
+        params, zenml_component_deploy
+    )
+
+    # writes the file names to the stack spec
+    # using format '<provider>-<component_name>.yaml'
+    for component in components:
+        stack.components.append(
+            f"{click.get_app_dir(MLSTACKS_PACKAGE_NAME)}/stack_specs/{stack.name}/{component.name}.yaml"
+        )
+
+    return stack, components
+
+
+@verify_installation
+def convert_mlstacks_primitives_to_dicts(
+    stack: "Stack", components: List["Component"]
+) -> Tuple[Dict[str, Any], List[Dict[str, Any]]]:
+    """Converts mlstacks Stack and Components to dicts.
+
+    Args:
+        stack: A mlstacks Stack object.
+        components: A list of mlstacks Component objects.
+
+    Returns:
+        A tuple of Stack and List[Component] dicts.
+    """
+    verify_mlstacks_prerequisites_installation()
+
+    # convert to json first to strip out Enums objects
+    stack_dict = json.loads(stack.json())
+    components_dicts = [
+        json.loads(component.json()) for component in components
+    ]
+
+    return stack_dict, components_dicts
+
+
+def generate_unique_filename(base_filename: str) -> str:
+    """
+    Generates a unique filename by appending a UUID to the base filename.
+
+    Args:
+        base_filename: The base filename.
+
+    Returns:
+        The unique filename.
+    """
+    unique_suffix = (
+        uuid.uuid4().hex
+    )  # Generate a random UUID and convert it to a string.
+    return f"{base_filename}_{unique_suffix}"
```

Comment:
not sure if this needs to be a separate method, essentially it just does `base_filename + "_" + str(uuid4())` right?

Reply:
Figured it might be something that could be reused somewhere else at some point.